src/hooks/useViolations.ts

```diff
@@ -45,27 +45,28 @@ const violationFields: Record<ViolationName, ViolationField> = {
     taxRequired: 'tax',
 };
 
-type ViolationsMap = Map<ViolationField, TransactionViolation[]>;
+type ViolationsMap = Map<MoneyRequestField, TransactionViolation[]>;
 
 function useViolations(violations: TransactionViolation[]) {
     const violationsByField = useMemo((): ViolationsMap => {
-        const violationGroups = new Map<ViolationField, TransactionViolation[]>();
-
-        for (const violation of violations) {
-            const field = violationFields[violation.name];
-            const existingViolations = violationGroups.get(field) ?? [];
-            violationGroups.set(field, [...existingViolations, violation]);
+        const filteredViolations = violations.filter((v) => v.type === 'violation');
+        const violationGroups = new Map<MoneyRequestField, TransactionViolation[]>();
+        for (const violation of filteredViolations) {
+            if (violation.type === 'violation') {
```

Comment:
We don't  need this here given we already did it for `filteredViolations`, right?